### PR TITLE
feat:Add keydown event to handle escape

### PR DESCRIPTION
**Pull request by CooperHash**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
#47887

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  add keydown event to handle escape key      |
| 🇨🇳 Chinese | 新增键盘事件来处理ESC         |

根据[mdn](https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values)的描述
image

popover弹出后，当点击esc后应该terminal，也就是回到关闭overlay回到原来的状态

改完前

https://github.com/ant-design/ant-design/assets/73218815/655d3431-a2bb-459d-8fa0-4d9dc48bb80f

更改后

https://github.com/ant-design/ant-design/assets/73218815/c9647881-1be0-429d-bc45-ae313c1a3131

p.s: 视频中的黑色弹出显示的是键盘按键esc

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

**Comment by CooperHash** [comment]
@afc163 pls take a look

**Comment by afc163** [comment]
You should move https://github.com/ant-design/ant-design/blob/ac18c99f92e273d16858861ef216d36999d19796/components/popconfirm/index.tsx#L81-L85 from Popconfirm to Popover.

**Comment by CooperHash** [comment]
which means handle this in popover, and delete it in popconfirm ?

**Comment by CooperHash** [comment]
and it seems `'keyCode' is deprecated.ts(6385)` 
should i still use it in popover

**Comment by CooperHash** [comment]
it is appropriate to write in this way? 
cause if i write `React.KeyboardEvent` it cannot use in `window.add...`

https://github.com/ant-design/ant-design/blob/48841c577ac535045952afd8a795e2711ae01f0d/components/popover/index.tsx#L56-L67

File: components/popover/index.tsx [hunk 1]
```diff
@@ -61,6 +77,8 @@ const Popover = React.forwardRef<TooltipRef, PopoverProps>((props, ref) => {
       prefixCls={prefixCls}
       overlayClassName={overlayCls}
       ref={ref}
+      visible={visible}
```
Comment (li-jia-nan) [inline components/popover/index.tsx:80]: `visible` is deprecated, please use `open` instead

File: components/popover/index.tsx [hunk 2]
```diff
@@ -50,6 +51,21 @@ const Popover = React.forwardRef<TooltipRef, PopoverProps>((props, ref) => {
 
   const overlayCls = classNames(overlayClassName, hashId, cssVarCls);
 
+  const [visible, setVisible] = React.useState(false);
```
Comment (li-jia-nan) [inline components/popover/index.tsx:54]:
```suggestion
  const [visible, setVisible] = React.useState(props.open ?? props.visible);
```

**Comment by li-jia-nan** [comment]
另外这个功能需要提交到 feature 分支

**Comment by afc163** [comment]
I mean Popconfirm already support this feature but Popover did not, we could move the exact code from Popconfirm to Popover.

**Comment by CooperHash** [comment]
> I mean Popconfirm already support this feature but Popover did not, we could move the exact code from Popconfirm to Popover.

i know
but popconfirm is use cloneElment children to make onKeydown event to popup

And on Popup dont have children 

It seems cannot use exact same code and logic.

If the code above I write (after fix default visible value) is ok (as you think),then I will close this and make a new pr to feature branch

**Comment by CooperHash** [comment]
#47928
i make a another same pr switch to feature branch, this can be close
@li-jia-nan

**Pull request closed without merging**